.github/workflows/ci.yml

```diff
@@ -0,0 +1,77 @@
+name: CI
+on:
+  push:
+    branches: [master, main]
+    tags: ["*"]
+  pull_request:
+
+concurrency:
+  # group by workflow and ref; the last slightly strange component ensures that for pull
+  # requests, we limit to 1 concurrent job, but for the master branch we don't
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ (github.ref != 'refs/heads/master' && github.ref != 'refs/heads/main') || github.run_number }}
+  # Cancel intermediate builds, but only if it is a pull request build.
+  cancel-in-progress: ${{ startsWith(github.ref, 'refs/pull/') }}
+
+env:
+  PYTHON: ~
```

Comment:
set the plot variable to true so we run the plotting tests too


just make sure that we do NOT install or use GLMakie. Originally this type of separation existed in the library from which you copied this because there the plotting tests were very expensive AND because installing GLMakie required a more expensive test runner server, so we had that filtered out and executed in only one of the CI runs